### DISCUSSION TITLE
How to add timestamp to logs?

### DISCUSSION THREAD

**Discussion by mariusa** [piku/piku#311] [Q&A] [answered]
Hi,

How can one add an ISO timestamp in front of each log entry?

Couldn't find a solution when searching the repo.

Thanks

**Comment by rcarmo** [comment, marked as answer]
Since many applications will do that themselves and the timestamps would be redundant, that is not done by design. As an example, Node-RED does this:

```
web.1 | 14 Jun 13:50:18 - [error] [sqlite:340ca86b4f6f67bf] Sqlite database not configured
web.1 | 14 Jun 13:50:18 - [warn] [RED.events] Deprecated use of "nodes-started" event from "/home/piku/.piku/envs/rss/node_modules/node-red-contrib-persist/persist.js:164:20". Use "flows:started" instead.
web.1 | 14 Jun 13:50:18 - [warn] [RED.events] Deprecated use of "nodes-started" event from "/home/piku/.piku/envs/rss/node_modules/node-red-contrib-persist/persist.js:164:20". Use "flows:started" instead.
web.1 | 14 Jun 13:50:18 - [warn] [RED.events] Deprecated use of "nodes-started" event from "/home/piku/.piku/envs/rss/node_modules/node-red-contrib-persist/persist.js:164:20". Use "flows:started" instead.
web.1 | 14 Jun 13:50:18 - [warn] [RED.events] Deprecated use of "nodes-started" event from "/home/piku/.piku/envs/rss/node_modules/node-red-contrib-persist/persist.js:164:20". Use "flows:started" instead.
```

There is [the `logdate` option](https://uwsgi-docs.readthedocs.io/en/latest/Options.html?highlight=logdate#logdate) to add a date time prefix to `uwsgi` subprocess unbuffered output (that can be added to [this block here](https://github.com/piku/piku/blob/master/piku.py#L1012)), but, again, it is not added by default since it is often redundant.
